dex/pair/src/fee.rs

```diff
@@ -109,9 +109,10 @@ pub trait FeeModule:
     }
 
     fn reinject(&self, token: &TokenIdentifier, amount: &BigUint) {
-        let mut reserve = self.pair_reserve(token).get();
-        reserve += amount;
-        self.pair_reserve(token).set(&reserve);
+        self.pair_reserve(token)
+            .update(|reserve| *reserve += amount);
+        self.pair_virtual_reserve(token, token)
```

Comment:
This is not needed - you need to redo the PR as you brought stuff from the main branch - which is not yet needed